components/listings-with-map.tsx

```
@@ -119,6 +132,113 @@ export default function ListingsWithMap() {
+    }
+
+    // Add commute time if specified
+    if (criteria.commute) {
+      // In a real application, this would use a routing service API
+      // For demo, we'll use a simple approximation
+      filtered = filtered.map(listing => ({
+        ...listing,
+        commuteTime: estimateCommuteTime(
+          listing.lat,
+          listing.lng,
+          criteria.commute!.to.coordinates.lat,
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)